### DISCUSSION TITLE
PS3 temperatures indicator for integrated PS2 emu

### DISCUSSION THREAD

**Discussion by kamilb880** [aldostools/webMAN-MOD#511] [Q&A]
Hi aldostools!

At first, I just wanted to say that your work for PS3 scene is amazing. You have really contributed and lots of people is grateful using your code daily :)

I've contacted you with some problem that is keeping my eyes awake for months now... And it is of course subject-mentioned temperature indicator within PS2 emu on PS3. 

I know that 3141card has implemented that solution to netemu (RSX/CELL temps visible in upper left corner of the screen after pressing PS button while playing PS2 game), but I'm happy owner of CECHE01, semi-BC console. Since my unit is using gxemu, I was not able to use temp mod implemented to netemu by 3141card (well, at least not without resigning from my GS usage).

I've contacted 3141card and he has confirmed that he have tried to mod ps2_emu and ps2_gxemu files without any success. We've tried another time, after many attempts we have achieved RAM dump on working gxemu.self, yet even 3141card has given up, as menu framework remained to be incomprehensible compared to netemu. If, by any chance, you'll be interested in Reverse Engineering it, I have all our work documented and stored.

Additionally, XDRAM of my semi-BC console has shown that there is a new sb controller, device 6 - compared to 3141card's PS3 slim. This surely is GS chip on my motherboard.

I was wondering if gxemu seems to be not moddable (due to not being able to identify menu views from RAM and process space dump, according to 3141card), is there another way of achieving temperature indicators?

I came across psx-place topic: "Is there a PS3 utility to show CPU/GPU temps while in game?" where you mention usage of http://ip-address-of-ps3/cpursx.ps3 - which seemd to me as perfect solution at first - to use another device to display my console temps. Yet, according to my knowledge Lv1 is reloaded when rebooting into ps2emu lpar, so we do not have access to cpursx.ps3 any more.

So, here are my questions:
- do you have any idea how to achieve temperature indicators over PS2 emu?
- do you have a hardware flasher, so we could patch the lv1.self?
- is there a possibility to read temperatures from another devices than CELL/RSX (sb_dev_05 and sb_dev_06)?
- are you interested in RE of gxemu.self? There's a lot work done already here, maybe we just need "fresh" eyes to look over this code

Thank you in advance for reply.

**Comment by aldostools** [comment]
@kamilb880 Thank you for your message. Unfortunately I don't have the experience that 3141card has with these kind of stuff.

I don't have idea of how to read the temperature in PS2 emu. When a PS2 is loaded, VSH is unloaded from LV2 (not LV1).
The syscalls and symbols used by PS2 emulator are different to the ones uses by GameOS. To find the additional symbols the PS2 needs to be reversed, and that requires time and experience in PPC assembler.

I don't have much experience with PPC assembler and I am very busy with my current projects (personal and from work).

Additionally my console is a slim model. So I only can experiment with ps2_netemu.

**Reply by kamilb880** [reply]
Aldostools, 

thank you for fast reply!
A shame though, that we'll not be able to drag this topic any further.
I keep my fingers crossed for your further work on PS3 scene!

Greetings
